daemon/daemon.go

```diff
@@ -63,19 +65,37 @@ func (ib *Infoblox) Allocate(args *ExtCmdArgs, result *types.Result) (err error)
 
 	log.Printf("RequestAddress: '%s', '%s', '%s'", netviewName, subnet, mac)
 	ip, _ := ib.Drv.RequestAddress(netviewName, subnet, "", mac, args.ContainerID)
-
+        log.Printf("Allocated IP: '%s'", ip)
+        hwAddr, err := hwaddr.GenerateHardwareAddr4(net.ParseIP(ip), hwaddr.PrivateMACPrefix)
```

Comment:
Ko, with CNI version v0.4.0, interface mac address is getting set using IP address. When request lands on ipam plugin we get dummy mac_addresses.After ipam plugin returns the ip address to the network_driver, it calls SetHwAddrByIp  function. Please have a look at bridge driver below:

IPAM call: https://github.com/containernetworking/cni/blob/v0.5.2/plugins/main/bridge/bridge.go#L252

SetHwAddrByIp call:
https://github.com/containernetworking/cni/blob/v0.5.2/plugins/main/bridge/bridge.go#L309

Same logic for other drivers also.